lib/extractTags.ts

```diff
@@ -0,0 +1,39 @@
+const typeGroup = /(\{([^\}]+)\})*/.source;
+const paramsGroup = /(\(([^\)]*)\))*/.source;
+const nameGroup = /([^\s]+)/.source;
+const nameWithDefaultValueGroup = /\[([^\s\]=]*)(=([^\]]+))?\]/.source;
+const descriptionGroup = /(?:-\s+([^\*\r\n]*))?/.source;
+const oneOrMoreWhiteSpace = /\s+/.source;
+const zeroOrMoreWhiteSpace = /\s*/.source;
```

Comment:
What is the purpose of using 'zero-or-more' expressions? 🤓 Sounds like a tautology that will evaluate always to true.